tests/unit_tests/mocks.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2022, Voltron Data.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
```

Comment:
This whole file makes me long for https://enpiar.com/r/httptest/ which AFAIK doesn't really have a python equivalent

Reply:
Ha, I sorta made this based on my experience with httptest. It's pretty nice.

`arrow-benchmarks-ci` [rolls their own](https://github.com/voltrondata-labs/arrow-benchmarks-ci/blob/b539ba6c7a282da63a57776cf7a56ccf4e44f6ea/envs/test/docker-compose.yml) mocking system. I loosely based this one off that, but drastically simplified.

Other python contenders are `pytest-mock` and `responses`, both of which I've used in the past, but since we're going the slightly-unconventional route of constructing and mounting our own HTTP adapter, sometimes mocking with them can be a bit tricky.